### PR TITLE
Map

### PR DESCRIPTION
**Pull request by MeuhV1**

### CODE REVIEW

**Review by dguenms — APPROVED** [review]
Recently I have also noticed that if you place a landmark that isn't associated with a player (i.e. with the world builder tool) and then save from WB you trigger an exception and the resulting file is in a corrupted state. I think it is because it is trying to create a player instance to write a comment into the file. It seems related to this, can you also look into it?

**Comment by MeuhV1** [comment]
I'll look into it.

**Comment by MeuhV1** [comment]
Got it to work. :D

File: Assets/Python/pyWB/CvWBDesc.py [hunk 1]
```diff
@@ -1809,7 +1809,10 @@ def write(self, f, sign):
 		f.write("BeginSign\n")
 		f.write("\tplotX=%d\n" %(sign.getPlot().getX(),))
 		f.write("\tplotY=%d\n" %(sign.getPlot().getY(),))
-		f.write("\tplayerType=%d, (%s)\n" %(sign.getPlayerType(), gc.getPlayer(sign.getPlayerType()).getName().encode(fileencoding)))
+		if sign.getPlayerType() == -1:
```
Comment (dguenms) [inline Assets/Python/pyWB/CvWBDesc.py]: Are you sure this works? Where is iPlayer defined?

**Comment by MeuhV1** [comment]
You're right about iPlayer not being defined. I found what caused the error and got it to work. After a successful test, I removed some testing code. Apperantly I removed 1 line too many. But this does fix it. (I tested it just before the push to be sure)

File: Assets/Python/pyWB/CvWBDesc.py [hunk 2]
```diff
@@ -1809,7 +1809,11 @@ def write(self, f, sign):
 		f.write("BeginSign\n")
 		f.write("\tplotX=%d\n" %(sign.getPlot().getX(),))
 		f.write("\tplotY=%d\n" %(sign.getPlot().getY(),))
-		f.write("\tplayerType=%d, (%s)\n" %(sign.getPlayerType(), gc.getPlayer(sign.getPlayerType()).getName().encode(fileencoding)))
+		iPlayer = sign.getPlayerType()
+		if sign.getPlayerType() == -1:
```
Comment (dguenms) [inline Assets/Python/pyWB/CvWBDesc.py]: This can now also be `iPlayer`

File: Assets/Python/pyWB/CvWBDesc.py [hunk 3]
```diff
@@ -1809,7 +1809,11 @@ def write(self, f, sign):
 		f.write("BeginSign\n")
 		f.write("\tplotX=%d\n" %(sign.getPlot().getX(),))
 		f.write("\tplotY=%d\n" %(sign.getPlot().getY(),))
-		f.write("\tplayerType=%d, (%s)\n" %(sign.getPlayerType(), gc.getPlayer(sign.getPlayerType()).getName().encode(fileencoding)))
+		iPlayer = sign.getPlayerType()
```
Comment (dguenms) [inline Assets/Python/pyWB/CvWBDesc.py:1812]: This looks weird on Github, as if there was an additional linebreak. Are there trailing whitespaces?

**Review by dguenms — CHANGES_REQUESTED** [review]

File: Assets/Python/pyWB/CvWBDesc.py (hunk 3 above)
Comment (MeuhV1) [reply Assets/Python/pyWB/CvWBDesc.py:1812]: The is no additional linebreak. There are also no trailing whitespaces. I guess it is just Github being weird.

I posted the new code in the post in the link below, so you can view it without Github peculiarities.
https://forums.civfanatics.com/threads/modmodding-q-a-thread.516755/page-94#post-15139548

EDIT:
Reuploaded to include iPlayer == -1, as you suggested.

**Review by dguenms — APPROVED** [review]

**Pull request merged**